client/components/EnhancementCard.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { Collection, CollectionItem, Row, Input, Button, Icon } from 'react-materialize';
+import ProductCard from './ProductCard.jsx';
+
+class EnhancementCard extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      userText: "",
+      dirty: false,
+    };
+
+    this.handleSearchText = this.handleSearchText.bind(this);
+    this.handleSelection = this.handleSelection.bind(this);
+  }
+
+  handleSearchText(e) {
+    this.setState({ userText: e.target.value, dirty: true })
+  }
+
+  handleSelection(e) {
+    this.props.handleEnhanceSelect(e);
+    this.setState({ userText: '', dirty: false });
+  }
+
+  render() {
+    const { enhancements, handleEnhanceSelect, selectedEnhancement, allEnhancementTags, selectedEnhancementTags, handleEnhancementFilter, removeFilter } = this.props;
+    const searchEnhancements = enhancements.filter(enhancement => enhancement.name.toLowerCase().match(this.state.userText.toLowerCase()));
```

Comment:
These start to become difficult to read on a narrower screen.

Some of us olds still use `vim` and like to pull up multiple files at once. A good rule of thumb is once you get to 80-100 characters, or more than 2-3 items in a list/collection, start breaking things over multiple lines.

More like this:
```js
const {
  enhancements,
  handleEnhanceSelect,
  selectedEnhancement,
  allEnhancementTags,
  selectedEnhancementTags,
  handleEnhancementFilter,
  removeFilter
} = this.props;
```